pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -460,7 +459,7 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation) *gardencorev1bet
 
 		destroyInternalDomainDNSRecord = g.Add(flow.Task{
 			Name:         "Destroying internal DNS entry",
-			Fn:           flow.TaskFn(component.OpWaiter(botanist.Shoot.Components.DNS.InternalEntry).Destroy),
+			Fn:           botanist.Shoot.Components.Extensions.DNS.InternalEntry.Destroy,
```

Comment:
Hm, but this will only delete the DNSEntry but won't wait until it is removed from store. Is it this change on purpose? I guess it is better to fail with `failed to wait until ... is deleted` instead of `failed to wait until Shoot namespace in the seed is deleted`.

Reply:
Hm, agreed, I wasn't aware of these implicit consequences. I'll change it back... thanks for finding it.